Style embedded links differently from normal links #225

MasterFelix15 opened the pull request:
Fixes issue: #225 

Staging site: https://rawgit.com/MasterFelix15/website/225-differentiate-embedded-links/index.html

The website now displays the embedded links in dark blue and in italic.

acjh commented:
Deferring to @damithc. I couldn't find a good reference on styling embedded/ajax links, so I'm open to this.

damithc commented:
I'm not very keen on using _italics_ because we use it for other things (e.g. when introducing a new term).
I remember some sites using a small arrow icon to indicate a link 'opens in new window'. Something similar is possible to indicate a link 'expands'?. Also, I think we use light gray dotted underlines to indicate tool tips. Can we use a variation of that to indicate this type of links?

MasterFelix15 commented:
@damithc @acjh I have changed the implementation to include a small icon which indicates that the link is expandable.

damithc commented:
> I have changed the implementation to include a small icon which indicates that the link is expandable.

Good work. How about special underline? Is that possible? If it is, we can also remove the blue color.

MasterFelix15 commented:
@damithc @acjh Added dotted underline and removed blue colour.
<img width="999" alt="screen shot 2017-03-05 at 11 49 52 pm" src="https://cloud.githubusercontent.com/assets/11973041/23601105/9bc77188-01fe-11e7-9014-fecf1b5ad04d.png">

damithc commented:
> I have changed the implementation to include a small icon which indicates that the link is expandable.

Looks good.

damithc commented on styles/common.css:
```diff
@@ -432,11 +432,16 @@ table.two-column-content td {
 .embedded-link {
-    color: blue;
-    text-decoration: underline;
+    color: black;
```
Perhaps we should let it keep the original color rather than override to black?

MasterFelix15 commented:
@damithc @acjh Colour changed to original blue.
<img width="1000" alt="screen shot 2017-03-06 at 9 21 25 am" src="https://cloud.githubusercontent.com/assets/11973041/23621154/69b712ca-024e-11e7-9a75-3361746de81c.png">

damithc commented:
> Colour changed to original blue

I meant retain the color of the text (i.e. don't override the color at all).

MasterFelix15 commented:
@damithc @acjh Sorry I misunderstood. The colour override has been cancelled.
<img width="997" alt="screen shot 2017-03-06 at 5 13 22 pm" src="https://cloud.githubusercontent.com/assets/11973041/23637382/40b8f78e-0290-11e7-8af9-c2c57dc13dbc.png">

damithc commented:
Looks good @MasterFelix15. A slight tweak would be to ensure the underline and the icon are the same color (grey, or light blue) but not strictly necessary.

MasterFelix15 commented:
@damithc @acjh The colour of the icon has been adjusted to grey to be consistent with the underline. 
<img width="998" alt="screen shot 2017-03-06 at 7 22 47 pm" src="https://cloud.githubusercontent.com/assets/11973041/23640427/7fd7c42e-02a2-11e7-8879-646aa2eef576.png">

acjh commented:
Hmm, the icon looks really pixelated. How about using CSS (sprite sheet) from FontAwesome (30 KB)? In comparison, the current icon is just 300 B.

Alternatives:
- jQuery UI (already loaded for ui-accordion's triangle)
- Bootstrap (more than 100 KB)

jQuery UI's icon seems a bit small (can try), while Bootstrap comes with a lot of unused styles (that I previously had problems using with jQuery UI).

damithc commented:
@MasterFelix15 any updates?

MasterFelix15 commented:
@damithc My apologies professor, I am having my exam week here in UCSD, can you allow me a few more days?

damithc commented:
> @damithc My apologies professor, I am having my exam week here in UCSD, can you allow me a few more days?

Sure.

MasterFelix15 commented:
@damithc @acjh Sorry for not being active for so long, I am having my exam week in UCSD, and it still isn't over yet... 

Regarding the previous problem, I looked into Fontawesome and there seem to be a small license issue if I want to use their icons. Besides, I do not think it is necessary to bring in an entire library just for that icon we want to use.

I have explored a bit and found out that there is really a hacky way of doing this: using the after selector in CSS and set the icon image as its background image. If we set the image as the content, we wouldn't be able to adjust the display size of the icon image, however, if we use it as a background image, we can do that by adjusting the background size.
<img width="1001" alt="screen shot 2017-03-19 at 11 50 43 pm" src="https://cloud.githubusercontent.com/assets/11973041/24090599/f4bda1ee-0cfe-11e7-9c2d-d026c78f8e39.png">

acjh commented on styles/common.css:
```diff
@@ -432,11 +432,23 @@ table.two-column-content td {
-    color: blue;
-    text-decoration: underline;
+    border-bottom: 1px dashed #999;
+    text-decoration: none;
```
Can remove `text-decoration` entirely (i.e. don't override too)

acjh commented on styles/common.css:
```diff
@@ -432,11 +432,23 @@ table.two-column-content td {
+    background-repeat: no-repeat;
+    background-position: 75% 75%;
+}
+
```
- Let's place this after `.embedded-link:active`
- Use `::after`
- Order properties alphabetically, `-webkit` first

acjh replied:
- `content` can be an empty `""`
- You can use `padding-left: 1ex;` for a little space before the icon

acjh replied:
@MasterFelix15 Any updates?

Also, the `-webkit` property is unnecessary.
Chrome 4.0 and Safari 4.1 (before which the `-webkit` prefix was used) were released in 2010.

MasterFelix15 commented:
@damithc @acjh Added small space before the icon, changed the space from 1ex to 0.5ex to look more comfortable. Cleaned up code as instructed, however, the background no-repeat element has to remain under ::after in order for it to work properly. And the padding-left has to be changed to margin-left since we are trying to control the background picture.
<img width="1000" alt="screen shot 2017-04-05 at 10 47 34 am" src="https://cloud.githubusercontent.com/assets/11973041/24687771/39a22c20-19ee-11e7-9aa4-fd8410d416e0.png">

acjh commented on styles/common.css:
```diff
@@ -432,13 +432,24 @@ table.two-column-content td {
+.embedded-link::after {
+    background: url("../images/expand.png");
+    background-size: 10px 10px;
+    background-repeat: no-repeat;
```
`background-repeat` before `background-size`

acjh commented on styles/common.css:
```diff
@@ -432,13 +432,24 @@ table.two-column-content td {
+
 .embedded-link:active {
     color: purple;
+    background-position: 75% 75%;
```
I think you misinterpreted this.
I meant moving the whole `.embedded-link::after` below `.embedded-link:active`.

That said, this shows that `background-position` is not necessary.
Could you also confirm that `background-repeat` is not necessary?

acjh commented:
@MasterFelix15 Any updates?
